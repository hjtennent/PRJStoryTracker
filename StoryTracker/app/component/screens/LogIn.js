import React, { useState } from "react"
import { View, TextInput, Text, Button } from "react-native"
import styles from "../styles/SignUp"
import auth from "@react-native-firebase/auth"

const LogIn = (props) => {
  const [email, setEmail] = useState("")
  const [password, setPassword] = useState("")
  const [errorMsg, setErrorMsg] = useState("")

  const handleLogIn = () => {
    auth().signInWithEmailAndPassword(email, password)
      .then(() => this.props.navigation.navigate('Home'))
      .catch(error => setErrorMsg(error.message))
  }

  return (
    <>
      <View style={styles.container}>
        <Text>Log In</Text>
        {errorMsg != "" && 
          <Text style={styles.errorMsg}>{errorMsg}</Text>
        }
        <TextInput 
          placeholder="Email"
          autoCapitalize="none"
          style={styles.textInput}
          onChangeText={(value) => setEmail(value)}
          value={email}
        />
        <TextInput 
          secureTextEntry
          placeholder="Password"
          autoCapitalize="none"
          style={styles.textInput}
          onChangeText={(value) => setPassword(value)}
          value={password}
        />
        <Button title="Log In" style={styles.signUpButton} onPress={() => handleLogIn()} />
        <View>
          <Text>Don't have an account? 
            <Text onPress={() => props.navigation.navigate('SignUp')} style={styles.loginLink}>
              Sign Up
            </Text>
          </Text>
        </View>
      </View>
    </>
  )

}

export default LogIn;