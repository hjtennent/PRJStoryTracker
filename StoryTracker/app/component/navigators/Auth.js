import React from 'react';
import LogIn from '../screens/LogIn';
import SignUp from '../screens/SignUp';
import { createStackNavigator } from '@react-navigation/stack';

export default Auth = () => {
  const Stack = createStackNavigator();
  return (
    <Stack.Navigator initialRouteName="Login">
      <Stack.Screen name="Login" component={LogIn} />
      <Stack.Screen name="SignUp" component={SignUp} />
    </Stack.Navigator>
  )
}