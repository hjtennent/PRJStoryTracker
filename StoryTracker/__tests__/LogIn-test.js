import React from 'react';
import renderer from 'react-test-renderer';
import LogIn from '../app/component/screens/LogIn';
import { NavigationContainer } from '@react-navigation/native';
import { fireEvent, render } from "@testing-library/react-native"
import Auth from '../app/component/navigators/Auth';

jest.mock('@react-native-firebase/auth', () => {
  return {
      onAuthStateChanged: jest.fn({user: 'user'}),
      signInWithEmailAndPassword: jest.fn(),
  }
})

test('renders correctly', () => {
  const tree = renderer.create(<LogIn />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Check email field is present', () => {
  const { getByPlaceholderText } = render(<LogIn />);
  const element = getByPlaceholderText('Email');
  expect(element).not.toBe(null);
})

test('Check password field is present', () => {
  const { getByPlaceholderText } = render(<LogIn />);
  const element = getByPlaceholderText('Password');
  expect(element).not.toBe(null);
})

//Check it navigates to SignUp
test('Check we can navigate to SignUp page', () => {
  const component = (
      <LogIn />
  );
  const { findByText } = render(component);
  const toClick = findByText('Sign Up');
  
  fireEvent(toClick, 'press');
  const signUpPageText = findByText('Already have an account?'); //specific to SignUp page

  expect(signUpPageText).toBeTruthy();
})

//Check LogIn button and firebase signin method
