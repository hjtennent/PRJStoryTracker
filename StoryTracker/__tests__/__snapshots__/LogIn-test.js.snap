// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    Log In
  </Text>
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    onChangeText={[Function]}
    placeholder="Email"
    rejectResponderTermination={true}
    style={
      Object {
        "borderBottomWidth": 1,
        "borderColor": "grey",
        "marginTop": 8,
        "width": "90%",
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    onChangeText={[Function]}
    placeholder="Password"
    rejectResponderTermination={true}
    secureTextEntry={true}
    style={
      Object {
        "borderBottomWidth": 1,
        "borderColor": "grey",
        "marginTop": 8,
        "width": "90%",
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Log In
      </Text>
    </View>
  </View>
  <View>
    <Text>
      Don't have an account?
      <Text
        onPress={[Function]}
        style={
          Object {
            "color": "red",
          }
        }
      >
        Sign Up
      </Text>
    </Text>
  </View>
</View>
`;
